trio/tests/test_util.py

```diff
@@ -58,3 +58,27 @@ def handler(signum, _):
                 async with ul1:
                     pass  # pragma: no cover
     assert "ul1" in str(excinfo.value)
+
+
+async def test_contextmanager_do_not_unchain_non_stopiteration_exceptions():
+    @acontextmanager
+    async def test_issue29692():
+        try:
+            yield
```

Comment:
Unfortunately we still support py35, so we can't use native generators like this in the test suite (this is why appveyor and travis are failing) – it needs to be like
```python
@async_generator
async def test_issue29692():
    try:
        await yield_()
    except ...
```